instant-xml-macros/src/de.rs

```diff
@@ -278,9 +313,15 @@ impl Deserializer {
             ));
         }
 
-        declare_values.extend(quote!(
-            let mut #enum_name: Option<#field_type> = None;
-        ));
+        if is_option_type {
+            declare_values.extend(quote!(
+                let mut #enum_name: Option<#field_type> = Some(None);
+            ));
+        } else {
+            declare_values.extend(quote!(
+                let mut #enum_name: Option<#field_type> = None;
```

Comment:
serde definitely just sets this to `None` for `Option` values, too.

Reply:
Yes, they have this function that handles that:
https://github.com/serde-rs/serde/blob/f52d134c14f6904010d0a59107987b050b36b811/serde/src/private/de.rs#L20